ark/ops/ops_sendrecv.cc

```diff
@@ -9,12 +9,12 @@ namespace ark {
 
 extern const OpConfigMap CommConfigMap;
 
-SendOp::SendOp(OpPrecType prec_type, Tensor *input, Tensor *output, int sid,
-               int rank, int dst_rank, size_t bytes, const std::string &name)
+SendOp::SendOp(OpPrecType prec_type, Tensor *input, int sid, int rank,
+               int dst_rank, size_t bytes, const std::string &name)
     : Op{OP_SEND,
          prec_type,
          {input},
-         {output},
+         {input},
```

Comment:
The interface change looks good to me. I am curious about why we need to put `input` as `output` in the Op constructor. Even though `send` doesn't need `output` buffer

Reply:
You are right, I think we can leave this argument empty here. Let me do some tests and update.

The following is just FYI: the current code still works correctly because actually, this argument is for passing an "output reference", not the actual output. Output reference points to where the op should write outputs, which is actually a kind of "input" of the op. The actual output tensor is created by the system, not provided by the user. So the current code is just passing a duplicated input which is still fine.

Reply:
I found that we actually need to provide `input` as an output reference. By design, current `Model::Impl::add_op()` need an output reference to create an output tensor.